### ISSUE TITLE
[Feature Request] rerun test on failures

### ISSUE THREAD

**Issue by hanayashiki** [lukeed/uvu#183]
Hi, I really like this library for its simplicity to use! No flamboyant encapsulation and extraneous stuff, it eases our testing like a breathe.

Currently, we're using it for browser tests, to e2e test our web application using puppeteer. E2E tests can fail for random reasons, and I think it should be helpful if we can add an optional retrial to this library.

`async-retry` can be a choice, but it is nicer if optional retrial can be integrated in the system, so that failures and retries can be displayed on the log along side with current data.

Thank you1

**Comment by alesmit** [comment]
> E2E tests can fail for random reasons

Tests generally have the goal to be deterministic, so if they yield different results every time they run the problem might have to be addressed on the test itself.

If your E2E tests are still flaky and you are using GH Actions on your project, I would suggest moving the retrial logic on the workflow with https://github.com/nick-invision/retry. I hope it helps until the author will provide some thoughts on this feature.

**Comment by hanayashiki** [comment]
> 

Yes, but that means I need to wrap every test with retry, which doesn't seem to be a good way. I'd be grateful if this could be configured globally
